pkg/readiness/pruner/pruner_test.go

```diff
@@ -184,3 +195,30 @@ func Test_ExpectationsMgr_DeletedSyncSets(t *testing.T) {
 		})
 	}
 }
+
+// Test_ExpectationsMgr_missedInformers verifies that the pruner can handle a scenario
+// where the readiness tracker's state will never match the informer cache events.
+func Test_ExpectationsMgr_missedInformers(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	// because we will use a separate lister for the tracker from the mgr client
+	// the contents of the readiness tracker will be superset of the contents of the mgr's client
+	tl := fakes.NewTestLister(
+		fakes.WithSyncSets([]*syncsetv1alpha1.SyncSet{
+			fakes.SyncSetFor("syncset-1", []schema.GroupVersionKind{podGVK, configMapGVK}),
+		}),
+	)
```

Comment:
Perhaps it's done elsewhere, but from reading the test it's not clear how there are two different data sources and (crucially) how their contents are different.

It would be ideal to have something like:

```
// Set up data store for readyTracker.  If you need to do this with imperative Create() calls that's fine. It is equivalent.
rtStore {
 syncSetA{gvk1, gvk2}
}

// Set up store for the controllers and watchManager
mainStore {
  syncSetA{gvk2}
  // As gvk1 is absent from SyncSetA, the CacheManager will never observe gvk1 being deleted and will never cancel the data expectation for gvk1
}
```

Reply:
tried to make it more obvious by following your lead here